lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -54,51 +54,46 @@ internal class AstToLogicalVisitorTransform(
-                                            lit(
-                                                projectItem.asAlias?.toIonElement()
-                                                    ?: errAstNotNormalized("SELECT-list item alias not specified")
-                                            ),
-                                            transformExpr(projectItem.expr),
-                                        )
-                                    is PartiqlAst.ProjectItem.ProjectAll -> {
-                                        structFields(transformExpr(projectItem.expr), projectItem.metas)
-                                    }
-                                }
-                            }
-                        )
-                    }
-                    is PartiqlAst.Projection.ProjectStar ->
-                        // `SELECT * FROM bar AS b` is rewritten to `SELECT b.* FROM bar as b` by
-                        // [SelectStarVisitorTransform]. Therefore, there is no need to support `SELECT *` here.
-                        errAstNotNormalized("Expected SELECT * to be removed")
-
-                    is PartiqlAst.Projection.ProjectPivot -> {
-                        problemHandler.handleUnimplementedFeature(node, "PIVOT")
-                        INVALID_EXPR
-                    }
-                },
-                algebra,
-                node.project.metas
-            )
-        }.let { q ->
-            // in case of SELECT DISTINCT, wrap bindingsToValues in call to filter_distinct
-            when (node.setq) {
-                null, is PartiqlAst.SetQuantifier.All -> q
-                is PartiqlAst.SetQuantifier.Distinct -> PartiqlLogical.build { call("filter_distinct", q) }
+            when (val project = node.project) {
+                is PartiqlAst.Projection.ProjectValue -> {
+                    bindingsToValues(
+                        exp = transformExpr(project.value),
+                        query = algebra,
+                        metas = project.metas
+                    )
+                }
+                is PartiqlAst.Projection.ProjectList -> {
+                    bindingsToValues(
+                        exp = transformProjectList(project),
+                        query = algebra,
+                        metas = project.metas
+                    )
+                }
+                is PartiqlAst.Projection.ProjectStar -> {
+                    // `SELECT * FROM bar AS b` is rewritten to `SELECT b.* FROM bar as b` by
+                    // [SelectStarVisitorTransform]. Therefore, there is no need to support `SELECT *` here.
+                    errAstNotNormalized("Expected SELECT * to be removed")
+                }
+                is PartiqlAst.Projection.ProjectPivot -> {
+                    pivot(
+                        input = algebra,
+                        key = transformExpr(project.key),
+                        value = transformExpr(project.value),
```

Comment:
The choice of key and value seems a bit unfortunate for me, but you don't have to change it.

Reply:
I agree, the naming is backwards. The unfortunate naming broke my brain a little haha, but unfortunately changing it is a breaking change.